modules/porous_flow/src/materials/PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.C

```diff
@@ -0,0 +1,75 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.h"
+
+template <>
+InputParameters
+validParams<PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity>()
+{
+  InputParameters params = validParams<PorousFlowTotalGravitationalDensityBase>();
+  params.addRequiredRangeCheckedParam<Real>(
+      "rho_s", "rho_s > 0", "The density of the solid matrix");
```

Comment:
I'm not sure if you might want
```
rho_s >= 0
```
here instead.  I know that in reality rho_s must be positive, but i've been caught in the past when people have suddenly wanted to do some comparison with analytical solutions where some quantity has been ignored so they've wanted to set it zero, and my range check has prevented them.  Probably i'm being overly cautious here - i'll let you decide.